common/nyxd-scraper/src/block_processor/mod.rs

```diff
@@ -44,6 +44,7 @@ impl PendingSync {
 
 pub struct BlockProcessor {
     pruning_options: PruningOptions,
+    store_precommits: bool,
```

Comment:
perhaps define an intermediate `Config` struct with all the possible options, like storing precommits, pruning, etc?